Hotcue fixes

poelzi opened the pull request:
Small updates to the hotcue overhaul:

* Fixes wrong placement of popup
* Wrong window size on first open
* Deletes hotcue on leftclick + shift
* Places the window on the left lower corner of the button
* Close Button on the window

poelzi committed:
Remove Hot-Cue on right click + shift

poelzi committed:
Fix wrong size of hotcue window

The size of the popup is used to calculate its placement.
Since the window was not yet shown, the layout is not caculated and
therfore the returned size is wrong. Force calculation on init.

poelzi committed:
Place the hotcue window left bottom of button

This looks much nicer then placing it randomly wherever the mouse is

Be-ing commented on src/widget/whotcuebutton.cpp:
```diff
@@ -89,6 +90,10 @@ void WHotcueButton::mousePressEvent(QMouseEvent* e) {
                 return;
             }
+            if (QGuiApplication::keyboardModifiers().testFlag(Qt::ShiftModifier)) {
+                pTrack->removeCue(pHotCue);
```
How about implementing this on the scrolling and overview waveforms too?

poelzi replied:
I implemented it in the overview. I implemented it shortly in the scrolling but then came the conclusion that it would make more sense to use this to move the cue point along the axis since this is high resolution and that would be cool if they could be moved.

Be-ing commented on src/widget/whotcuebutton.cpp:
```diff
@@ -89,8 +89,13 @@ void WHotcueButton::mousePressEvent(QMouseEvent* e) {
+            }
             m_pCueMenuPopup->setTrackAndCue(pTrack, pHotCue);
-            m_pCueMenuPopup->popup(e->globalPos());
+            // use the lover left corner as starting point for popup
```
```suggestion
            // use the bottom left corner as starting point for popup
```

poelzi committed:
Implement delete cue on rightclick + shift in overview

poelzi committed:
Update hotcue tooltip

Be-ing commented on src/skin/tooltips.cpp:
```diff
@@ -597,12 +598,19 @@ void Tooltips::addStandardTooltips() {
+                  << QString("%1 + %2: %3")
+                             .arg(rightClick,
+                                     shift,
+                                     tr("Delete selected cuepoint."));
```
```suggestion
                                     tr("Delete selected hotcue."));
```

poelzi replied:
mmhh, when we have saved loops, this could also be a cue loop right ?

poelzi replied:
I renamed it for now

Be-ing commented on src/util/widgethelper.cpp:
```diff
@@ -17,18 +17,21 @@ QPoint mapPopupToScreen(
     }
-    const auto screenSize = pWindow->screen()->size();
+    // the screen geometry is the physical screen of the virtual desktop
+    // this will be offset by it's top and left starting points
```
```suggestion
    // this will be offset by its top and left starting points
```

Be-ing commented on src/widget/wcuemenupopup.cpp:
```diff
@@ -42,6 +42,12 @@ WCueMenuPopup::WCueMenuPopup(UserSettingsPointer pConfig, QWidget* parent)
     m_pDeleteCue->setObjectName("CueDeleteButton");
     connect(m_pDeleteCue, &QPushButton::clicked, this, &WCueMenuPopup::slotDeleteCue);
 
+    // FIXME(XXX): icon instead of text ?
```
Isn't there a standard icon you could use for this? Maybe `QIcon::fromTheme("window-close")`?

Holzhaus replied:
Note that this might return an invalid icon if the system's icon theme cannot be determined or lacks this icon, so we need to fall back to text in that case.

poelzi replied:
We can alter skin the icon from the qss

Holzhaus commented:
> * Close Button on the window

Can you elaborate what this is useful for? You can just click somewhere else to close it. Other right-click menus like WTrackMenu don't have a close button either.

poelzi commented:
@Holzhaus I know, it just feels wrong to have no close button there for this kind of windows. It is not very intuitive that clicking somewhere outside will close it. I also have to search of a unused place where it is safe to click. This button is small and does not use up any space that would not be there anywhere (space is created by the delete button). So having a dedicated button to close the window is from a ui perspective the right thing to do.

Be-ing commented:
I'm ambivalent about the addition of the close button. I don't think it's necessary, but I suppose it doesn't hurt.

Be-ing commented:
There is a merge conflict now.

Holzhaus commented:
> I'm ambivalent about the addition of the close button. I don't think it's necessary, but I suppose it doesn't hurt.

I'm not a fan of it. The cue menu is a right-click context menu, not a window. It auto-closes if you click a color button for example.

Other right-click menus like `WTrackMenu` don't have a close button  either. Either we introduce inconsistency, or we add close buttons to these menus either.

Be-ing commented:
> Other right-click menus like WTrackMenu don't have a close button either. Either we introduce inconsistency, or we add close buttons to these menus either.

That is a good point. It is odd for a right click menu to have a close button.

poelzi committed:
Fix wrong screen detection

In multi screen desktops, the check uses the wrong size for detecting
if the window will be on the screen.
No it looks for the physical screen mixxx window is at and moves the
window inside this monitor.

poelzi committed:
spelling, simplify code

poelzi commented:
I removed the close button. ready for merge
